app/shared/design-system/components/auto-input.tsx

```diff
@@ -0,0 +1,129 @@
+export class AutoInput extends EventTarget {
+	connectedCallback() {
+		if (!this.activeInput) this.activeInput = null;
+		if (!this.canvasContext) Object.getPrototypeOf(this).canvasContext = document?.createElement('canvas')?.getContext('2d');
+		for (let name of ['keydown', 'mouseup', 'mousedown', 'mousemove']) globalThis.addEventListener(name, this, false);
```

Comment:
afaik mousemove is a very performance sensitive event, because triggered a lot, you want to only bind it after mousedown and unbind it on mouseup